bindings/jaxrs/src/main/java/org/commonjava/indy/core/bind/jaxrs/VersioningTestHandler.java

```diff
@@ -0,0 +1,148 @@
+ * limitations under the License.
+ */
+package org.commonjava.indy.core.bind.jaxrs;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import org.commonjava.indy.bind.jaxrs.IndyResources;
+import org.commonjava.indy.core.ctl.StatsController;
+import org.commonjava.indy.stats.IndyVersioning;
+import org.commonjava.indy.util.ApplicationContent;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Response;
+
+import static org.commonjava.indy.bind.jaxrs.util.ResponseUtils.formatOkResponseWithJsonEntity;
+
+/**
+ * We do the following steps when we release a new apiVersion, e.g. 2
+ *
+ *  1) Update '<apiVersion>1 -> 2</apiVersion>' in indy-parent pom.xml. It will auto-update apiVersion in
+ *  'indy-version.properties' in indy-model-core-java during build. This apiVersion will be released along with the
+ *  client in one of the dependency jars.
+ *
+ *  2) If an api is to be changed in terms of req/resp message formats or the way how it handles the requests
+ *  and we want to support both v1 and v2 users, we copy the current method and create a new method name
+ *  (e.g., getIndyVersion_v1) and annotate it with '@Produce "application/indy-v1+json"'. This method serves old v1
+ *  client users.
+ *
+ *  3) We continue working on the v2 version of this api with all the updates via the original method name
+ *  (e.g., getIndyVersion). This method serves new v2 client users.
+ *
+ *  4) If no changes to an api, we just leave it as is. All v1 and v2 clients will accessing it by default.
+ *  (e.g., getIndyInfo)
+ *
+ *  ## Something behind this versioning strategy:
+ *
+ *  When client sends a requests to Indy, it loads the apiVersion in the 'indy-version.properties' in the
+ *  client jar (e.g., 1) and add a header 'Indy-API-Version' in the request (e.g., 'Indy-API-Version: 1'). When server
+ *  receives the request, it goes through ApiVersioningFilter to adjust 'Accept' header with the most acceptable MIME
+ *  type being prioritised. The servlet container respect the priorities and choose the right method to access.
+ *
+ */
+@Api( description = "For versioning test only.", value = "Versioning Test" )
+@Path( "/api/test/versioning" )
+public class VersioningTestHandler
+                implements IndyResources
+{
+
+    private final Logger logger = LoggerFactory.getLogger( getClass() );
+
+    @Inject
+    private StatsController statsController;
+
+    @Inject
+    private ObjectMapper objectMapper;
+
+    @ApiOperation( "Retrieve version information about this Indy instance" )
+    @ApiResponse( code = 200, response = IndyVersioning.class, message = "The version metadata" )
+    @Path( "/ver" )
+    @GET
+    @Produces( "application/indy-v1+json" )
```

Comment:
The whole VersioningTestHandler is for test. I come up with the mock API endpoints and 2 cases (method upgraded, method no change) and make sure both the old client and new client can access them. Maybe I need to change the method names so that they do not confuse.